furbysource.asm

```diff
@@ -506,4 +506,1073 @@ InitIO:
+; If light osc fails, or too dark and that sends us to sleep, we
+; set 'Dark_sleep_prev' and save it in EEPROM in 'Seed_2'.
+; when the sleep routine executes,(00 01 based on this bit)
+; When we wake up we recover this bit and it becomes the previous done
+; flag back in 'Stat_0', so that if the osc is
+A-23
+; still dark or failed, Furby wont go back to sleep.
+      LDA   Seed_2            ;from EEPROM
+      BED   No_prevsleep      ;jump if none
+      LDA   Stat_0            ;system
+      ORA   #Dark_sleep_prev ;prev done
+      STA   Stat_0            ;update
+
+No_prevsleep:
+
+;***************************************************************
+
+      LDA   Spcl_seed1  ;recover start up random number
+      STA   Seed_1            ;set generator
+
+;***************************************************************
+
+
+; Pot_timeL2 is save in ram through sleep mode and then reloaded '
+; Pot_timeL which is the working register for the motor position.
+; This allows startup routines to clear ram without forgetting the
+; last motor position.
+
+      LDA   Pot_timeL2  ;get current count
+      STA   Pot_ imeL   ;save in motor routine counter
+
+;*************************
+
+; Get age and make sure it is not greater than 3 (age4)
+
+      LDA   Age         ;get current age
+      AND   #83h    ;preserve bit 7 which is 9th age counter bit
+;;;;;             and insure age not >3
+
+      STA   Age         ;set system.
+
+;*************************
+
+      LDA   #Bored_reld ;reset timer
+      STA   Bored_timer ;
+
+      LDA   #03         ;set timer
+      STA   Last_IR           ;timer stops IR from hearing own IR xmit
+      
+      JSR   Get_light   ;go get light level sample
+      LDA   TEMP1       ;get new count
+      STA   Light_reff  ;update system
+
+
+;**********************************************************************
+*
+
+      LDA   Warm_cold   ;decide if warm or cold boot
+      CMP   #11h        ;ck for warm boot
+      BEQ   No_zero           ;jump if is
+A-24
+      LDA   #00         ;point to macro 0 (SENDS TO SLEEP POSITION)
+      STA   Macro_Lo
+      STA   Macro_Hi
+      JSR   Get_macro   ;go start motor/speech
+      JSR   Notrdy            ;Do / get status for speech and motor
+
+No_zero:
+
+
+      LDA   #11         ;preset motor speed
+      STA   Mon_len           ;set motor on pulse timing
+
+      LDA   #05        ;set motor to 3/4 speed for speed test
+      STA   Moff_len   ;set motor off pulse timing
+;
+;
+      LDA   #00        ;clear all system sensor requests
+      STA   Stat_4           ;update
+
+
+; Currently uses 4 tables, one for each age.
+      LDA   Stat_0            ;system
+      ORA   #Init_motor :flag motor to do speed test
+      ORA   #Init_Mspeed      ;2nd part of test
+      STA   Stat_0            ;update
+
+
+;************************************************************
+
+; Do wake up routine :
+
+      lda   #Global_time      ;reset timer to trigger sensor learning
```

Comment:
Making a note: spirit would capitalize `lda`, exactness would leave it to mimic the document